contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,419 @@
+        """
+        Returns a Space object
+        """
+        return Box(np.array(joint_angle_gripper_cmd_lower), np.array(joint_angle_gripper_cmd_upper))
+        # TODO
+        # Add other control modes
+        # make it more precise
+
+    @property
+    def observation_space(self):
+        """
+        Returns a Space object
+        """
+        return self._observation_space
+
+    # Functions that gazebo env asks to implement
+    #-----------------------------------------------------------------------------------
+    def _env_setup(self, initial_qpos):
+        # Generate the world
+        # Load the model
+        GazeboEnv.load_gazebo_sdf_model(model_name="cafe_table",
+                              model_pose=Pose(position=Point(initial_qpos['table0'][0],
+                                                             initial_qpos['table0'][1],
+                                                             initial_qpos['table0'][2])),
+                              model_path=rospkg.RosPack().get_path('sawyer_sim_examples') + "/models/" + "cafe_table/model.sdf")
+        GazeboEnv.load_gazebo_urdf_model(model_name="block",
+                               model_pose=Pose(position=Point(initial_qpos['object0'][0],
+                                                              initial_qpos['object0'][1],
+                                                              initial_qpos['object0'][2])),
+                               model_path=rospkg.RosPack().get_path('sawyer_sim_examples') + "/models/" + "block/model.urdf")
+
+        self._move_to_start_position()
+
+        # Extract information for sampling goals
+        gripper_pose = self._limb.endpoint_pose()
+        GazeboEnv.load_gazebo_sdf_model(model_name="target",
+                              model_pose=Pose(position=Point(gripper_pose['position'][0],
+                                                             gripper_pose['position'][1],
+                                                             gripper_pose['position'][2])),
+                              model_path=rospkg.RosPack().get_path('sawyer_sim_examples') + "/models/" + "target/model.sdf")
+        self.initial_gripper_pos = np.array(gripper_pose['position'])
+        self.initial_gripper_ori = np.array(gripper_pose['orientation'])
+        if self.has_object:
+            self.height_offset = initial_qpos['object0'][2]
+
+    def _sample_goal(self):
+        """
+        Sample goals
+        :return:
+        """
+        print("Sampling new goal...")
+
+        if self.has_object:
+            goal = self.initial_gripper_pos[:3] + self.np_random_generator.uniform(-self.target_range,
+                                                                                   self.target_range, size=3)
+            goal[2] = self.height_offset
+            if self.target_in_the_air and self.np_random_generator.uniform() < 0.5:
+                goal[2] += self.np_random_generator.uniform(0, 0.45)
```

Comment:
You should write a comment here why at 50% of the cases you want to offset the goal's z position.